src/Quahog/Result.php

```diff
@@ -0,0 +1,96 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Xenolope\Quahog;
+
+class Result
+{
+    private const RESULT_OK    = 'OK';
+    private const RESULT_FOUND = 'FOUND';
+    private const RESULT_ERROR = 'ERROR';
+
+    /**
+     * @var string
+     */
+    private $id;
+
+    /**
+     * @var string
+     */
+    private $filename;
+
+    /**
+     * @var string|null
+     */
+    private $reason;
+
+    /**
+     * @var string
+     */
+    private $status;
+
+    public function __construct(string $status, string $filename, ?string $reason, ?string $id)
+    {
+        $this->status   = $status;
+        $this->filename = $filename;
+        $this->reason   = $reason;
+        $this->id       = $id;
+    }
+
+    /**
+     * @return string
+     */
+    public function getId(): ?string
+    {
+        return $this->id;
+    }
+
+    /**
+     * @return string
+     */
+    public function getFilename(): string
+    {
+        return $this->filename;
+    }
+
+    /**
+     * @return null|string
+     */
+    public function getReason(): ?string
+    {
+        return $this->reason;
+    }
+
+    /**
+     * @return bool
+     */
+    public function ok(): bool
```

Comment:
I think these `bool` method names would be a little clearer if they had the standard `is` prefix, e.g. `isOk`, otherwise it's not super clear if you're setting or getting a value (I know this is a value object, but consumers might not).

Reply:
If the `is` prefix is prefered, I'll add them. No problem. I was more puzzled about the other namings using `get`.

* `isOk`
* `hasFailed`?
* `isFound` or `wasFound`?
* `isError`

Reply:
These look great now, thanks!